controllers/numaresourcesoperator_controller.go

```diff
@@ -334,8 +339,40 @@ func (r *NUMAResourcesOperatorReconciler) deleteUnusedDaemonSets(ctx context.Con
 				if err := r.Client.Delete(ctx, &ds); err != nil {
 					klog.ErrorS(err, "error while deleting daemonset", "DaemonSet", ds.Name)
 					errors = append(errors, err)
+				} else {
+					klog.V(3).Infof("DS[%s] deleted", ds.Name)
+				}
+			}
+		}
+	}
+	return errors
+}
+
+func (r *NUMAResourcesOperatorReconciler) deleteUnusedMachineConfigs(ctx context.Context, instance *nropv1alpha1.NUMAResourcesOperator, mcps []*machineconfigv1.MachineConfigPool) []error {
+	klog.V(3).Info("Delete MC start")
+	var errors []error
```

Comment:
Is there a special reason why errors are a slice?
Usually, if we're hitting an error we should return immediately

Reply:
I thought this and PR #76  where intended as an improvement to delete unused resources but not as a primary task for the operator, that's why I thought in do it in "best-effort" in case of error and try to delete as much resources as possible not stopping at the first error. 
Also an error on this function or the one deleting DaemonSets do not make the sync function to return as an error.

@fromanirh was also not too sold on this either...

Reply:
yes, I'm still not sold, but I don't fight against it either. I'm open to small experiments. However, If we decide to change, let's change everything in one go.

The reason I'm sold is that doing this way breaks the expected flow, so one stops and pauses and in the back of their head the "why is this different?" thought starts. It breaks the flow.
Obviously some cases _are_ special enough to break the rules. This case doesn't strike me as obviously special, yet.

Reply:
I mean I just did it that way just because I understood the initiative as a best-effort to delete unused resources, but maybe I misunderstood the intention and then it is better to change it to keep things homogeneous.

Reply:
no worries, it's ok. Let's try this approach and we will have a more informed opinion for the future.